src/HotChocolate/Fusion/src/Core/Planning/RequestFormatters/RequestDocumentFormatter.cs

```diff
@@ -83,15 +85,37 @@ internal abstract class RequestDocumentFormatter(FusionGraphConfiguration config
     }
 
     private SelectionSetNode CreateRootLevelQuery(
+        QueryPlanContext context,
         SelectionPath path,
         SelectionSetNode selectionSet)
     {
         var current = path;
 
         while (current is not null)
         {
-            selectionSet = new SelectionSetNode(
-                new[] { current.Selection.SyntaxNode.WithSelectionSet(selectionSet), });
+            var selectionNode = current.Selection.SyntaxNode.WithSelectionSet(selectionSet);
+
+            // TODO: this is not good but will fix the include issue ...
+            // we need to rework the operation compiler for a proper fix.
+            if (selectionNode.Directives.Count > 0)
```

Comment:
Whats the issue with the resolver compiler? I am working btw on a separate branch that will no longer use the operation compiler in fusion.

Reply:
I just copied your comment from [here](https://github.com/ChilliCream/graphql-platform/blob/9a08ac26af7baed29f351e266debf9fffe055c86/src/HotChocolate/Fusion/src/Core/Planning/RequestFormatters/RequestDocumentFormatter.cs#L214) :D